### PR TITLE
ARC-1506-audience-tech-planning-adjusting-dbt-arc-functions-util-create-or-update-standard-models-to-account-for-replicating-snapshot-config-files

### PR DESCRIPTION
**Pull request by ryantimjohn**
# Pull Request for dbt-arc-functions

- [x] Open the pull request as a draft so that you can run tests against the code and inspect the files in the Files view of the Pull Request interface in Github.

- [x] Once you're satisfied, convert the draft into an open Pull Request by clicking "Ready for Review" near the bottom of this page. You should not make any changes to the code except in dire emergency once you've converted from draft, so make sure you're really satisfied.

- [x] Add two of the principal contributors to this PR for review. Currently, the principal contributors are @dmbluestate , @Frydafly , and @ryantimjohn.

- [x] We've divided our PR templates into Macros, Utils and Docs PR requests. Delete the ones you're not using.
### Macros (delete if not using)

- [x] Link Jira tickets (or Github Issues) to PR here. Make sure that the Jira ticket has a good description as to why this PR is necessary and the business problem that this PR is solving. Moreover, if there have been any discussions of this ticket in Slack, make sure they're linked in the Jira ticket:
https://bluestate.atlassian.net/browse/ARC-1506

- [x] Make sure this is named using Jira extension so that naming convention is consistent. If it's not, that's okay. Go to the Jira ticket, click into the ticket, then on the right under Details, find Development, then click the Create branch button. Copy the Branch Name from there. Read here [how to rename a branch in Github](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch) to rename this branch. In the future, create branches from Jira.

- [x] Make sure the Macro is documented by running the notebook: `create_docs.ipynb` util in this repo. From there, you'll have to either fill in descriptions by hand OR use our `fill_in_descriptions_using_openai.ipynb` to fill in descriptions using OpenAI's ChatGPT. Always make sure to check over ChatGPT's column descriptions, as they're likely not perfect.

- [x] Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in `packages.yml` file in dbt State client below.
done on UUSA:
https://cloud.getdbt.com/develop/1713/projects/207950
Snapshot runs successfully:
Screenshot 2023-06-23 at 3 46 19 PM

- [x] check in dbt that client package successfully runs `dbt deps`, `dbt compile`, and `dbt run`; screenshot this page
successfully run on UUSA
Screenshot 2023-06-23 at 3 48 19 PM
Screenshot 2023-06-23 at 3 48 08 PM

- [x] query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this
Screenshot 2023-06-23 at 3 49 47 PM

### Utils (delete if not using)

- [x] Suggest any tests that your PR reviewer might use to check the utils here. Better yet, if you're able, write automated tests in Github Actions!
I would suggest running `python create_or_update_standard_models.py` from the utils folder against a client you know, and selecting `stitch_sfmc_arc` model, which contains the audience snapshot. Moreover, I'd also throw some other models in there to make sure this works.

- [x] Update utils/README.md with any new utils. Make sure that the Table of Contents is updated (most IDEs/code editors can do this for you!)

### CODE REVIEW

**Comment by ryantimjohn** [comment]
@dmbluestate @Frydafly  can see this in action here: https://github.com/bsd/arc-uusa-dbt/pull/10

File: standard_snapshots/stitch_sfmc_arc/snapshots/snp_stitch_sfmc_arc_audience.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+version: 2
+snapshots:
+  - name: snp_stitch_sfmc_arc_audience
```
Comment (Frydafly) [inline standard_snapshots/stitch_sfmc_arc/snapshots/snp_stitch_sfmc_arc_audience.yml:4]: just noting difference between this name and @dmbluestate 

obviously you both worked on it concurrently and separately, but before merging both, would be nice to have one name for this snapshot!

File: standard_snapshots/stitch_sfmc_arc/snapshots/snp_stitch_sfmc_arc_audience.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{% snapshot snp_stitch_sfmc_arc_audience %}
```
Comment (Frydafly) [inline standard_snapshots/stitch_sfmc_arc/snapshots/snp_stitch_sfmc_arc_audience.sql:1]: same here of course

File: utils/create_or_update_standard_models.py [hunk 3]
```diff
@@ -498,10 +517,15 @@ def create_or_update_readme(dbt_models_path, list_of_sources):
...
+    standard_paths = [
+        os.path.join(dbt_models_path, model_type) for model_type in model_types
+    ]
+    for standard_path in standard_paths:
+        try:
+            set_of_existing_models.update(os.listdir(standard_path))
+        except FileNotFoundError:
+            print(f"{standard_path} directory does not exist, moving on.")
     list_of_existing_models = list(set_of_existing_models)
     list_of_existing_models.sort
```
Comment (Frydafly) [inline utils/create_or_update_standard_models.py:530]: hola, just noticed this when I opened up this branch on my visual studio code: missing () after sort

**Review by Frydafly — COMMENTED** [review]
Other than ideally addressing the differences between naming conventions here and naming conventions used in https://github.com/bsd/arc-uusa-dbt/pull/8 (however you and @dmbluestate see fit to resolve),  looks good to me

**Review by Frydafly — APPROVED** [review]

File: standard_snapshots/stitch_sfmc_arc/snapshots/snp_stitch_sfmc_arc_audience.yml (hunk 1 above)
Comment (dmbluestate) [reply standard_snapshots/stitch_sfmc_arc/snapshots/snp_stitch_sfmc_arc_audience.yml:4]: @ryantimjohn @Frydafly I actually prefer to defer to tim's usage here. After testing this code, I will adapt the other code to use this naming.

**Review by dmbluestate — APPROVED** [review]
Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in packages.yml file in dbt State client below.
done on ARC_UUSA:
Snapshot runs successfully:
Screenshot 2023-06-26 at 2 54 52 PM




 check in dbt that client package successfully runs dbt deps, dbt compile, and dbt run; screenshot this page
successfully run on ARC_UUSA:
dbt deps
Screenshot 2023-06-26 at 2 55 40 PM

dbt compile

Screenshot 2023-06-26 at 2 55 53 PM

dbt run 
Screenshot 2023-06-26 at 2 56 12 PM




 query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this

Screenshot 2023-06-26 at 2 57 59 PM

File: utils/create_or_update_standard_models.py (hunk 3 above)
Comment (ryantimjohn) [reply utils/create_or_update_standard_models.py]: good catch!

**Pull request merged**